src/cmem/ContentBlobToggler/StringPreviewContentBlobToggler.tsx

```diff
@@ -39,14 +40,15 @@ export function StringPreviewContentBlobToggler({
     noTogglerContentSuffix,
 }: StringPreviewContentBlobTogglerProps) {
     const previewMaybeFirstLine = firstNonEmptyLineOnly ? firstNonEmptyLine(content) : content;
-    const previewString = previewMaxLength ? previewMaybeFirstLine.substr(0, previewMaxLength) : previewMaybeFirstLine;
-    const enableToggler = previewString !== content;
+    const enableToggler = (previewMaxLength ?? Infinity) < content.length;
     let previewContent = renderPreviewAsMarkdown ? (
-        <Markdown key="markdown-content" allowedElements={allowedHtmlElementsInPreview}>
-            {previewString}
-        </Markdown>
+        <TextReducer maxLength={previewMaxLength}>
```

Comment:
`TextReducer` usage does not make sense at this point because it does the exact opposite to `renderPreviewAsMarkdown`, it removes any markup created by the Markdown renderer component.